Fix support for custom operators returning std::tuple and taking c10::optional

jamesr66a opened the pull request:
Summary: These are bindings for ops functionally equivalent to Caffe2's SparseLengthsSum/SparseLengthsWeightedSum and LengthsRangeFill, respectively.

Differential Revision: D15125484

soumith commented:
spoke with james about this offline. We'll push stuff like this into a separate extension that's private / internal, rather into pytorch core -- because it doesn't have plans for frontend integration yet.

Commit:
_lengths_range and _embedding_bag_lengths experimental operator bindings (#19926)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/19926

These are bindings for ops functionally equivalent to Caffe2's SparseLengthsSum/SparseLengthsWeightedSum and LengthsRangeFill, respectively.

Differential Revision: D15125484

fbshipit-source-id: 415543945a751265bb776bb4299938c4c3750d1a
